biomage/experiment/download.py

```diff
@@ -0,0 +1,306 @@
+            _download_file(bucket, s3_path, file_path)
+
+        print(f"Sample {sample_name} downloaded.\n")
+
+    _create_sample_mapping(samples_list, output_path)
+    click.echo(
+        click.style(
+            "All samples for the experiment have been downloaded.",
+            fg="green",
+        )
+    )
+
+
+def _download_rds_file(experiment_id, input_env, output_path, processed=False):
+    file_name = None
+    bucket = None
+
+    if not processed:
+        file_name = "raw_r.rds"
+        bucket = f"{RAW_FILES_BUCKET}-{input_env}"
+        end_message = "Raw RDS files have been downloaded."
+    else:
+        file_name = "processed_r.rds"
+        bucket = f"{PROCESSED_FILES_BUCKET}-{input_env}"
+        end_message = "Processed RDS files have been downloaded."
+
+    key = f"{experiment_id}/r.rds"
+    file_path = output_path / file_name
+
+    _download_file(bucket, key, file_path)
+
+    print(f"RDS file saved to {file_path}")
+    click.echo(click.style(f"{end_message}", fg="green"))
+
+
+def _download_cellsets(experiment_id, input_env, output_path):
+    FILE_NAME = "cellsets.json"
+
+    bucket = f"{CELLSETS_BUCKET}-{input_env}"
+    key = experiment_id
+    file_path = output_path / FILE_NAME
+    _download_file(bucket, key, file_path)
+    print(f"Cellsets file saved to {file_path}")
+    click.echo(click.style("Cellsets file have been downloaded.", fg="green"))
+
+
+@click.command()
+@click.option(
+    "-e",
+    "--experiment_id",
+    required=True,
+    help="Experiment ID to be copied.",
+)
+@click.option(
+    "-i",
+    "--input_env",
+    required=True,
+    help="Input environment to pull the data from.",
+)
+@click.option(
+    "-o",
+    "--output_path",
+    required=False,
+    default=None,
+    help="Name of output folder. By default this will be the experiment id.",
+)
+@click.option(
+    "-a",
+    "--all",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Download all files for the experiment.",
+)
+@click.option(
+    "--name_with_id",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Use sample id to name samples.",
+)
+@click.option(
+    "-f",
+    "--files",
+    multiple=True,
+    required=False,
+    default=[SAMPLES],
+    help=(
+        "Files to download. By default only the samples (-f samples) are downloaded. "
+        "You can also download cellsets (-f cellsets), raw RDS (-f raw_rds) and "
+        "processed RDS (-f processed_rds)."
+    ),
+)
+def download(experiment_id, input_env, output_path, files, all, name_with_id):
+    """
+    Downloads files associated with an experiment from a given environment.\n
+
+    E.g.:
+    biomage experiment download -i staging -e 2093e95fd17372fb558b81b9142f230e
+    -f samples -f cellsets -o output/folder
+    """
+
+    # Set output path
+    # By default, the output path is named after the experiment id
+    if not output_path:
+        output_path = experiment_id
+
```

Comment:
this if is redundant -- `output_path` will always be defined, because you have it at the top of the file as a global variable.

Reply:
`output_path` is actually a command input (`-o`). If it's not given, then we'll be using the experiment id (which is done in this block), else we use the path that is supplied